python/oneflow/nn/modules/global_cast.py

```diff
@@ -75,16 +77,135 @@ def global_to_global_op(
 
 
 def to_global_op(input, placement=None, sbp=None, **kwargs):
-    assert isinstance(input, Tensor)
-
-    if input.is_global:
-        return global_to_global_op(input=input, placement=placement, sbp=sbp, **kwargs)
+    r"""Converts the input tensor or input tensor(s) in list/tuple/dict to global tensor(s).
+    
+    It performs the same conversion as :func:`oneflow.Tensor.to_global` to the tensor or every tensor in the input.
+
+    Note:
+        Both placement and sbp are required if the input is local, otherwise at least one of placement and sbp is required.
+
+    Args:
+        input (flow.Tensor/None/list/tuple/dict): the input that needs to be converted.
+        placement (flow.placement, optional):  the desired placement of the input. Default: None
+        sbp (flow.sbp.sbp or list/tuple of flow.sbp.sbp, optional): the desired sbp of the input. Default: None
+    
+    Returns:
+        The converted input.
+
+    For a tensor input: please refer to the examples in :func:`oneflow.Tensor.to_global`.
+
+    For an input of other type (take a state dict as an example):
+
+    .. code-block:: python
+
+        >>> # Run on 2 ranks respectively
+        >>> import oneflow as flow
+        >>> from oneflow import nn
+        >>> placement = flow.placement("cpu", ranks=[0, 1]) # doctest: +SKIP
+        >>> sbp = (flow.sbp.broadcast,) # doctest: +SKIP
+        >>> model = nn.Sequential(nn.Linear(8, 4), nn.ReLU(), nn.Linear(4, 2)) # doctest: +SKIP
+        >>> global_state_dict = flow.to_global(model.state_dict(), placement, sbp) # doctest: +SKIP
+        >>> for val in state_dict.values(): # doctest: +SKIP
+        >>>     print(val.is_global) # doctest: +SKIP
+
+    .. code-block:: python
+
+        >>> # results on rank 0
+        True
+        True
+        True
+        True
+
+    .. code-block:: python
+
+        >>> # results on rank 1
+        True
+        True
+        True
+        True
+    """
+    def _to_global_tensor(input_tensor, placement=None, sbp=None, **kwargs):
```

Comment:
move outside this function to avoid def at every call

Reply:
Corrected